tests/Controller/Api/GalleryControllerTest.php

```diff
@@ -124,6 +124,9 @@ public function testGetGalleryMediaAction()
         $this->assertSame([$media], $gController->getGalleryMediasAction(1));
     }
 
+    /**
+     * @group legacy
+     */
```

Comment:
@supersmile2009 please review. Is it right to mark this as legacy? Are we testing the legacy path here? Are there tests for the non-legacy path?

Reply:
For what I've seen the test is creating a mock gallery object that implements `GalleryInterface` which in fact has the deprecated method `addGalleryHasMedias` that makes the test not pass.

There is an if...else in `GalleryController::handleWriteGalleryhasmedia` that checks if it has to call `addGalleryHasMedias` (deprecated) or `addGalleryHasMedia` (the new method) depending on the interface the gallery implements (`GalleryInterface` or `GalleryMediaCollectionInterface`).

```
// NEXT_MAJOR: remove this if/else block. Just call `$gallery->addGalleryHasMedia($galleryHasMedia);`
if ($gallery instanceof GalleryMediaCollectionInterface) {
    $gallery->addGalleryHasMedia($galleryHasMedia);
} else {
    $gallery->addGalleryHasMedias($galleryHasMedia);
}
$this->galleryManager->save($gallery);
```

So if the test is testing a legacy path it should be marked as legacy.
Otherwise if the test is intended to test the new `addGalleryHasMedia` we should change the mock gallery object to be an instance of `GalleryMediaCollectionInterface` so the if in `GalleryController` runs `addGalleryHasMedia`. I'm not sure how to do this.

May be the solution for now is to leave the test as legacy an once the `addGalleryHasMedias` method is removed (along with the if in the controller) also remove the legacy tag because it will call the new `addGalleryHasMedia` method.

Reply:
Ideally there should be 2 tests, but your changes can be merged IMO